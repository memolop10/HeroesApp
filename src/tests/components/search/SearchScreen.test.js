import { mount } from "enzyme";
import { MemoryRouter, Route } from "react-router-dom";
import SearchScreen from "../../../components/search/SearchScreen";

describe('Test in <SearchScreen />', () => {
    test('should render correctly with values by default', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Route path="/search" component={ SearchScreen }/>
            </MemoryRouter>
        )

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.form-control').exists() ).toBe(true);
        expect( wrapper.find('.alert-info').text().trim() ).toBe('Search a Hero');
    })

    test('should show batman and the input with the value of the queryString', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route path="/search" component={ SearchScreen }/>
            </MemoryRouter>
        )
        
        expect( wrapper.find('input').prop('value') ).toBe('batman');    
        expect( wrapper ).toMatchSnapshot();    
    })
    
    test('should show an error if not found the hero', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman36']}>
                <Route path="/search" component={ SearchScreen }/>
            </MemoryRouter>
        )
    
        expect( wrapper.find('.alert-danger').exists() ).toBe(true);    
        expect( wrapper ).toMatchSnapshot();  
    })
    
    test('should called the push of history', () => {
        const history ={ 
            push: jest.fn()
        }

        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route 
                    path="/search" 
                    component={ () => <SearchScreen history={ history }/> }
                />
            </MemoryRouter>
        )

        wrapper.find('input').simulate('change',{
            target: {
                name:'search',
                value:'batman'
            }
        });

        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        });

        expect( history.push ).toHaveBeenCalledWith(`?q=batman`)


    })
    
});
