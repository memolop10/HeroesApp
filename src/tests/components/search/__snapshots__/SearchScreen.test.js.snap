// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchScreen /> should render correctly with values by default 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a Hero
      </div>
    </div>
  </div>
</div>
`;

exports[`Test in <SearchScreen /> should show an error if not found the hero 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman36"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        There not a heroe with 
        batman36
      </div>
    </div>
  </div>
</div>
`;

exports[`Test in <SearchScreen /> should show batman and the input with the value of the queryString 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        className="card ms-3 mb-2"
        style={
          Object {
            "backgroundColor": "#0D142D",
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/heroes/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                <small>
                  More...
                </small>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
